specutils/io/default_loaders/jwst_reader.py

```diff
@@ -119,6 +146,52 @@ def jwst_x1d_multi_loader(file_obj, **kwargs):
     return _jwst_x1d_loader(file_obj, **kwargs)
 
 
+@data_loader("JWST x1d MIRI MRS", identifier=identify_jwst_miri_mrs,
+             dtype=SpectrumList, extensions=['*'])
+def jwst_x1d_miri_mrs_loader(input, **kwargs):
+    """
+    Loader for JWST x1d MIRI MRS spectral data in FITS format
+
+    A single data set consists of a bunch of _x1d files corresponding to
+    a variety of wavelength bands. This reader reads one by one and packs
+    the result into a SpectrumList instance.
+
+    Parameters
+    ----------
+    input: list with str or file-like
+          List of FITS file names, or objects (provided from name by
+          Astropy I/O Registry). Alternatively, a directory path on
+          which glob.glob runs with pattern an implicit pattern "_x1d.fits",
+          or a directory path with a glob pattern already set.
+
+    Returns
+    -------
+    SpectrumList
+        A list of the spectra that are contained in all the files.
+    """
+    # If input is a list, go read each file. If directory, glob-expand
+    # list of file names.
+    if not isinstance(input, list):
+        if os.path.isdir(input):
+            file_list = glob.glob(os.path.join(input, "*_x1d.fits"), recursive=True)
+        else:
+            file_list = glob.glob(input, recursive=True)
+
+    else:
+        file_list = input
+
+    spectra = []
+    for file_obj in file_list:
+        sp =  _jwst_x1d_loader(file_obj, **kwargs)
```

Comment:
The specutils loaders raise `FileNotFound` when encounter that situation. I think we should preserve the same behavior here.

Reply:
The situation is slightly different here as you want to accept wildcards (or even more so, with a directory as input, when this would fail once there is any file present that is not a valid FITS spectrum, right?)
Maybe needs some more discussion how strict this should be (or add an option to fail or pass on unreadable files.

Reply:
@dhomeier that's why it uses a glob pattern when accessing a directory with nothing else specified.

Reply:
Ah, I see, so it only iterates through anything ending on `_x1d.fits`. Maybe that pattern could be customised (independently from offering the full path name pattern as `input`), but I don't know how strictly those naming conventions are enforced with JWST files.

Reply:
@dhomeier yes, I think there is room in here to discuss other possible choices and combinations for the input parameters that specify the files to be read. For one, the `_x1d.fits` pattern is hard-coded, but should be a keyword parameter instead (defaulting to `_x1d.fits`). Passing a path name pattern is part of the requirements, but doesn't work without more extensive work with the existing subclassing/delegation architecture of `SpectrumList`.

Reply:
Loading one file, throwing error is fine. But what if you have 100 files, do you really want it to error out on the last invalid file after loading the other 99? Or do you want it to continue on anyway with a warning?

Reply:
I'll let this decision to the higher-ups. I don't know much about specutils' requirements regarding these kinds of situations.

Reply:
I think this will be the first example where a single call would access many files.
Yes, best let the maintainers chime in; I'd repeat my suggestion to add a kwarg option, perhaps defaulting to `warn` with choices of "raise" and "silent".

Reply:
I'm in favor of a kwarg option as you recommended @dhomeier. I think I'm going to commit @pllim's suggestion as a starting point and then iterate off of that with some additional suggestions.